In [17]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

print("Apps duplicadas:", df.duplicated(subset=['App']).sum())

df.drop_duplicates(subset=['App'], keep='first', inplace=True)

print("Apps duplicadas después de la purga:", df.duplicated(subset=['App']).sum())

Apps duplicadas: 1181
Apps duplicadas después de la purga: 0


- Pregunta A: El scraper cometió errores y guardó la misma app varias veces. Escriban una línea de código para identificar si hay nombres duplicados en la columna App (df.duplicated(subset=['App']).sum()). Si los hay, apliquen .drop_duplicates(subset=['App'], keep='first', inplace=True) para quedarse solo con la primera aparición de cada aplicación. ¿Cuántos registros duplicados purgaron?

Purgamos 1181 registros duplicados.

In [18]:
df['Installs'] = df['Installs'].str.replace('+', '', regex=False)
df['Installs'] = df['Installs'].str.replace(',','', regex=False)
df['Installs'] = df['Installs'].str.replace('Free','0', regex=False).astype(float)

df['Installs'].mean()

np.float64(7776701.607349897)

- Pregunta B: La columna Installs tiene texto como "1,000,000+". Un algoritmo no puede procesar eso.
    - Escriban una instrucción encadenando el método .str.replace() para eliminar el símbolo + y otro para eliminar la coma ,. (Ojo: utilicen regex=False o escapen el símbolo más \+).
    - Al final de la cadena, utilicen .astype(int) (o float si tienen NaNs) para convertir la columna.
    - Comprueben el éxito ejecutando df['Installs'].mean(). ¿Cuál es el promedio real de descargas?

Tambien se tuvo que reemplazar los valores 'Free' por ceros. El promedio de descargas fue de 7776701.6073

In [19]:
df['Price'] = df['Price'].str.replace('$','', regex=False)
df['Price'] = df['Price'].str.replace('Everyone','0', regex=False).astype(float)

- Pregunta C: La columna Price tiene valores como "$4.99" y "0". Apliquen la misma lógica arquitectónica que en la Pregunta B: eliminen el símbolo de dólar $ y conviertan la columna usando .astype(float).

Reemplazé también el valor 'Everyone' por ceros. 

In [22]:
df['Price'].max()

df[df['Price'] > 200]

df = df[df['Price'] < 50]

df.to_csv('playstore_limpio.csv', index=False)

- Pregunta D: Ahora que el precio es numérico, ejecuten df['Price'].max(). Notarán que la app más cara cuesta alrededor de 400 dólares.
    - Usen un filtro booleano (df[df['Price'] > 200]) para descubrir cuáles son estas apps. Verán que son aplicaciones basura o de broma.
    - Si metemos estas apps al modelo, la red neuronal creerá que vender a 400 dólares es un modelo de negocio viable. Sobrescriban el DataFrame conservando únicamente las apps que cuesten menos de 50 dólares (df = df[df['Price'] < 50]).
    - Guarden su dataset final usando df.to_csv('playstore_limpio.csv', index=False).